internal/node/node.go

```diff
@@ -30,7 +30,14 @@ func NewNode(client client.Client) Node {
 	}
 }
 
-func (n *node) IsNodeSchedulable(node *v1.Node) bool {
+func (n *node) IsNodeSchedulable(node *v1.Node, tolerations []v1.Toleration) bool {
+	for _, toleration := range tolerations {
```

Comment:
If the node has 2 taints, but we only added one toleration to the KMM Module, then the node will be deemed schedulable, although it should be so, no?

Reply:
I agree, we are only satisfied (in the current PR) with one fitting taint<-->toleration while what we need is to make sure that **all** the taints are tolerated.

Reply:
@yevgeny-shnaidman I have updated the matching criteria. Please review. Thanks